### PR TITLE
Add AssetSentimentRepository and AssetSentimentController

### PR DESCRIPTION
**Pull request by jackstenglein**
Adds the AssetSentimentRepository and AssetSentimentController to the server, as well as tests for the controller. Currently, only the getAssets function is added to the controller. Future functionality will be added in the next pull request.

### CODE REVIEW

File: server/src/main/java/com/google/moviestvsentiments/assetSentiment/Asset.java [hunk 1]
```diff
@@ -72,6 +72,26 @@ public int hashCode() {
...
+     * Creates a new Asset with the provided id, type and title.
+     * @param assetId The id of the new Asset.
+     * @param assetType The type of the new Asset.
+     * @param title The title of the new Asset.
+     * @return A new Asset with the provided id, type and title.
+     */
+    public static Asset create(String assetId, AssetType assetType, String title) {
+        return new Asset(assetId, assetType, title);
+    }
+
```
Comment (TianyiMu) [inline server/src/main/java/com/google/moviestvsentiments/assetSentiment/Asset.java:94]: Given the Spring requirement, would it be better to just use the public constructor here?

File: server/src/main/java/com/google/moviestvsentiments/assetSentiment/AssetSentimentController.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+     */
+    @GetMapping("/assets")
+    public List<AssetSentiment> getAssets(@RequestParam("assetType") AssetType assetType,
+                                          @RequestParam("accountName") String accountName,
+                                          @RequestParam("sentimentType") SentimentType sentimentType) {
+        Iterable<AssetSentiment> withReaction = assetSentimentRepository.getAssetsWithSentiment(assetType,
+                accountName, sentimentType);
+
+        if (sentimentType != SentimentType.UNSPECIFIED) {
+            return StreamSupport.stream(withReaction.spliterator(), false).collect(Collectors.toList());
```
Comment (TianyiMu) [inline server/src/main/java/com/google/moviestvsentiments/assetSentiment/AssetSentimentController.java:36]: I see both methods in AssetSentimentRepository gives List, why do we need to stream and collect to list again here?

File: server/src/main/java/com/google/moviestvsentiments/assetSentiment/Asset.java (hunk 1 above)
Comment (TianyiMu) [reply server/src/main/java/com/google/moviestvsentiments/assetSentiment/Asset.java:94]: Also for UserSentiment.java

Comment (jackstenglein) [reply server/src/main/java/com/google/moviestvsentiments/assetSentiment/Asset.java]: I just removed the changes and put the functionality directly in the test.

**Pull request merged**